### PR TITLE
Allow admin to disable background ajax calls on global search until user hits enter

### PR DESCRIPTION
**Pull request by scottshipman**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

On systems with large amounts of data, having the global search kickoff background search suggestions on 3 typed chars can take a while. This allows an admin to toggle a setting on the System configuration page to disable the background ajax calls until the user hits enter.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Set the "Disable Search Until Enter" flag to True in system configurations.
3. from any page that contains the global search widget, type 1-3 characters, including a space, and verify no search ajax request occurs.
4. Type 4 or more characters and verify no ajax search request occurs.
Hit Enter, verify search ajax request occurs and a suggestion modal pops up (standard behavior).

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.

### CODE REVIEW

File: app/bundles/CoreBundle/Config/config.php [hunk 1]
```diff
@@ -1085,16 +1085,17 @@
...
-        'link_shortener_url'        => null,
-        'cached_data_timeout'       => 10,
-        'batch_sleep_time'          => 1,
-        'batch_campaign_sleep_time' => false,
-        'cors_restrict_domains'     => true,
-        'cors_valid_domains'        => [],
-        'rss_notification_url'      => 'https://mautic.com/?feed=rss2&tag=notification',
-        'max_entity_lock_time'      => 0,
-        'default_daterange_filter'  => '-1 month',
+        'do_not_track_internal_ips'     => [],
```
Comment (kuzmany) [inline app/bundles/CoreBundle/Config/config.php]: To many spaces? PHPStorm? :)

**Review by kuzmany — DISMISSED** [review]
I like this PR
Tested on Mautibox, works properly.
Code looks legit for me.

**Comment by kuzmany** [comment]
Are you using this in production?

**Comment by npracht** [comment]
Hi there!
It has been decided to not create any extra Mautic 2 minor versions (which means no more features in M2) knowing that Mautic 3 is coming very soon.

**We now want to integrate your contribution in the Mautic 3 roadmap as 3.1.0 candidate.**

## How to do?
1. Check if your feature or enhancement is still missing and relevant in Mautic 3
2. **If Yes**: rebase your PR against the 3.x branch
3. **If No**: please consider closing your PR if you are absolutely sure that the feature has made it into Mautic 3 already

**Please report results by commenting on your PR to make us administration easier.**
______
You can more information on how to do all of that on [this blog post "Getting you PR ready for Mautic 3"](https://www.mautic.org/blog/developer/getting-your-pr-ready-mautic-3).

**Comment by npracht** [comment]
Hi @scottshipman ! Is this still relevant for M3 ?

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@scottshipman @kuzmany it would be nice to get this into 3.2 if we can, I think it would be a useful feature to have, particularly on large instances.

It seems we just need to update the config file to include the more recent additions in order for this to be considered and for @scottshipman to sign the contributors agreement (https://www.mautic.org/contributor-agreement) - then we should be good to go.

**Pull request closed without merging**